### ISSUE TITLE
Host to Docker connections fail when Tailscale incoming connections are enabled

### ISSUE THREAD

**Issue by nsmith22** [tailscale/tailscale#5393]
### What is the issue?

I have a Docker service running via Docker Compose, which exposes port 3003 for HTTP requests. I also have the "Accept incoming connections" feature enabled in Tailscale. When I stand up the Docker service, I can successfully make HTTP requests to it from my browser on the Host (i.e. Chrome can load http://localhost:3003). However, after a few minutes, any attempt to make HTTP requests from my host results in a connection timeout.

I tried the same steps with Tailscale incoming connections disabled and could not reproduce.

In both scenarios, however, I can access the service on port 3003 from _within_ the Docker network. For example, if I have a second service spun up with Docker Compose, that service can make HTTP requests to the other one regardless of the Tailscale setting.

What I later found that is definitely causing this that I have an HAProxy server in AWS on my Tailscale network which uses my laptop as a backend server, and importantly the HAProxy configuration has the `check` option on. With this option on, "HAProxy will try to establish a TCP connection every two seconds" (https://www.haproxy.com/blog/how-to-enable-health-checks-in-haproxy/). Turning this `check` option off fixes the problem, though it should not be necessary. It seems that some sort of network resource is being exhausted due to the continuous connection attempts, possibly due to improper resource cleanup.

### Steps to reproduce

1. Enable the Tailscale "Allow incoming connections" feature.
2. Use Docker Compose to spin up a simple HTTP server which exposes a port to the host.
3. Ensure that you can make HTTP requests from the host to the container service on the exposed port via `localhost:<port>`.
4. Set up an HA proxy server on another machine in the Tailscale network to use your machine's IP address as a backend, like this:
```
backend my_web_server
  server xyz <machine_ip>:<port> check
```
6. Let this sit for 5-10 minutes such that HAProxy is continuously making connections to your machine.
7. Try to make the same HTTP request as in step 3, and you should experience a connection timeout after some period of time.

### OS

macOS

### OS version

Monterey 12.1

### Tailscale version

1.28.0

### Bug report

BUG-b3dfb312371ceed137c9a79abada85bd1a0fe3b284970894ac9bedbf99e045c8-20220817195445Z-2b18ee7ea8944b40

**Comment by DentonGentry** [comment]
Unfortunately I didn't look at the bugreport when this first appeared.

I wondered if the idle connections from HAProxy were not being closed, but we added keepalives in 1.24.1 for that reason: https://github.com/tailscale/tailscale/issues/4522#issuecomment-1111595369

**Comment by DentonGentry** [comment]
At this point I don't expect to diagnose what was happening last August. If it happens again and a fresh bugreport is attached, we can look into it.
